util/dvsim/FpvCfg.py

```diff
@@ -0,0 +1,166 @@
+        results_str = "## " + self.results_title + "\n\n"
+        results_str += "### " + self.timestamp_long + "\n"
+        results_str += "### FPV Tool: " + self.tool.upper() + "\n\n"
+
+        header = ["Name", "Result"]
+        colalign = ("center", ) * len(header)
+        table = [header]
+        counts = {"proven": 0, "cex": 0, "undetermined": 0, "covered": 0, "unreachable": 0}
+        summary = ""
+        cov_header = ["Stimuli", "COI", "Proof"]
+        cov_table  = [cov_header]
+
+        for mode in self.build_modes:
+            result_data = Path(
+                subst_wildcards(self.build_dir, {"build_mode": mode.name}) +
+                '/results.csv')
+            try:
+                with open(result_data, "r") as results_file:
+                    reader = csv.reader(results_file)
+                    self.result = dict((rows[0], rows[1]) for rows in reader)
+                    for key in self.result:
+                        table.append([key, self.result[key]])
+                        counts[self.result[key]] = counts.get(self.result[key]) + 1
+                    total_tests = counts.get("proven") + counts.get("cex") + \
+                                  counts.get("undetermined")
+                    if total_tests == 0:
+                        counts["PassRate"] = "0.00%"
+                    else:
+                        counts["PassRate"] = "{:.2%}".format(counts.get("proven") / (total_tests))
+                    total_covs = counts.get("covered") + counts.get("unreachable")
+                    if total_covs == 0:
+                        counts["CovRate"] = "0.00%"
+                    else:
+                        counts["CovRate"] = "{:.2%}".format(counts.get("covered") / (total_covs))
+                os.remove(result_data)
+
+            except IOError as err:
+                log.error("%s", err)
+                results_str += str(err)
+
+            if (self.cov):
+                cov_data = Path(
+                  subst_wildcards(self.build_dir, {"build_mode": mode.name}) +
+                  '/cov.csv')
+                try:
+                    with open(cov_data, "r") as results_file:
+                        reader = csv.reader(results_file)
+                        cov_result = dict((rows[0], rows[1]) for rows in reader)
+                        cov_percentage = []
+                        for cov_name in cov_header:
+                          if cov_name in cov_result:
+                            cov_percentage.append(cov_result[cov_name])
+                        cov_table.append(cov_percentage)
+                    os.remove(cov_data)
+                except IOError as err:
+                    log.error("%s", err)
+                    results_str += str(err)
+
+        if table != [header]:
+            self.results_md = results_str + tabulate(
+                table, headers="firstrow", tablefmt="pipe", colalign=colalign)
+        else:
+            self.results_md = results_str
```

Comment:
The performance issue is due to the "tabulate" function here. When modules like `top_earlgrey` has thousands of properties, each property is an item in the tabulates. This blows up the calculation. I am going to only print out the summary.